backend/mem/file.go

```diff
@@ -0,0 +1,444 @@
+package mem
+
+import (
+	"errors"
+	"io"
+	"path"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//File implements vfs.File interface for the in-memory implementation of FileSystem.
+type File struct {
+	exists       bool
+	lastModified time.Time
+	isOpen       bool
+	contents     []byte       //the file contents
+	name         string       //the base name of the file
+	cursor       int          //the index that the buffer (contents) is at
+	location     vfs.Location //the location that the file exists on
+}
+
+/*		******* Error Functions *******		*/
+func doesNotExist() error {
+	return errors.New("This file does not exist!")
+}
+
+func copyFailNil() error {
+	return errors.New("The target file passed in was nil")
+}
+
+func seekError() error {
+	return errors.New("Seek could not complete the desired call")
+}
+
+/*		******************		*/
+
+//Close imitates io.Closer by resetting the cursor and setting a boolean
+func (f *File) Close() error {
+
+	if f == nil {
+		return errors.New("Cannot close a nil file")
+	}
+
+	f.isOpen = false
+	f.cursor = 0
+	return nil
+}
+
+//Read implements the io.Reader interface.  Returns number of bytes read and potential errors
+func (f *File) Read(p []byte) (n int, err error) {
+
+	//if file exists:
+	offset := f.cursor
+	if f.isOpen == false {
+		f.isOpen = true
+	}
+	existence, err := f.Exists()
+	if !existence {
+		return 0, err
+	}
```

Comment:
Seems like this should be more along the lines of:
`exists, err := f.Exists() // using 'exists' vs 'existence' because it makes more sense grammatically in conditionals ie: 'if [it] exists' vs 'if [it] existence'
//handle Exists errors first
if err != nil {
    return 0, err
}
if !exists {
    return 0, nil // not sure offhand whether it would be more consistent with other implementations to error for read on non-existent file or not.
}`

Reply:
well, that formatted badly.